siliconcompiler/apps/sc.py

```diff
@@ -45,12 +45,12 @@ def main():
     if chip.get('design') == UNSET_DESIGN:
         topfile = None
         for sourceset in ('rtl', 'hll'):
-            if chip.valid('input', sourceset):
-                for filetype in chip.getkeys('input', sourceset):
-                    sources = chip.get('input', sourceset, filetype)
-                    if sources:
-                        topfile = sources[0]
-                        break
+            for filetype in chip.getkeys('input', sourceset):
+                sources = chip.schema._getvals('input', sourceset, filetype)
+                if sources:
+                    # triple index to get first of this structure: ([val], <step>, <index>)]
+                    topfile = sources[0][0][0]
```

Comment:
Could you break this into two accesses into the array? Something like files=sources[0] and topfile = files[0][0] that way the comment is a little more clear.

Reply:
fixed, thanks - I should know better :)